tls/extensions/s2n_server_key_share.c

```diff
@@ -35,26 +34,134 @@ const s2n_extension_type s2n_server_key_share_extension = {
     .if_missing = s2n_extension_noop_if_missing,
 };
 
-int s2n_extensions_server_key_share_send_check(struct s2n_connection *conn);
+static int s2n_server_key_share_generate_pq_hybrid(struct s2n_connection *conn, struct s2n_stuffer *out) {
+    notnull_check(out);
+    notnull_check(conn);
+
+    ENSURE_POSIX(s2n_is_in_fips_mode() == false, S2N_ERR_PQ_KEMS_DISALLOWED_IN_FIPS);
 
-static int s2n_server_key_share_send(struct s2n_connection *conn, struct s2n_stuffer *out)
-{
-    GUARD(s2n_extensions_server_key_share_send_check(conn));
+    struct s2n_kem_group_params *server_kem_group_params = &conn->secure.server_kem_group_params;
+
+    notnull_check(server_kem_group_params->kem_group);
+    GUARD(s2n_stuffer_write_uint16(out, server_kem_group_params->kem_group->iana_id));
+
+    struct s2n_stuffer_reservation total_share_size = { 0 };
+    GUARD(s2n_stuffer_reserve_uint16(out, &total_share_size));
+
+    struct s2n_ecc_evp_params *server_ecc_params = &server_kem_group_params->ecc_params;
+    notnull_check(server_ecc_params->negotiated_curve);
+    GUARD(s2n_stuffer_write_uint16(out, server_ecc_params->negotiated_curve->share_size));
+    GUARD(s2n_ecc_evp_generate_ephemeral_key(server_ecc_params));
+    GUARD(s2n_ecc_evp_write_params_point(server_ecc_params, out));
+
+    /* s2n_kem_send_ciphertext() expects to find the client's public key in the
+     * params passed to it. We temporarily point server_kem_params->public_key to
+     * the correct location. */
```

Comment:
Nit: I'd move this comment down a few lines to just after `notnull_check(client_kem_params->public_key.data);`